### ISSUE TITLE
[Feature]: Space, backspace, tab, and back-tab the whole selection

### ISSUE THREAD

**Issue by gituser1000000** [AndroidIDEOfficial/AndroidIDE#1229]
### Feature description

Pasted text many times does not get formatted correctly. Either that or we might just want to manually adjust it. I can't even tell how many times I have tediously tried to space and tab every single line after pasting a long text.

Being able to move the whole selection at the same time would make things so much easier.

### What version of AndroidIDE you're using?

v2.5.2 (debug builds)

### Duplicate issues

- [X] This issue has not been reported yet.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by itsaky** [comment]
I'm working on this. This is how the feature would work :

- User selects the text
- User presses the `TAB` button (either in keyboard or the symbol input bar)
- The selected text's indentation is increased by one.

This behavior would be good with a `space` character, and I don't know what you mean by `back-tab` (remove indentation?).

**Comment by gituser1000000** [comment]
> ... and I don't know what you mean by back-tab (remove indentation?).

Yes, I just called it that (I had to call it something that would correspond to a backspace) since we don't have a key for it on a keyboard, and typically just use the normal backspace key to remove a tab character.

If we could indent the selection, it's reasonable that we should also to be able to reverse the effect, de-indenting, right?

> This behavior would not be good with a space character, ...

It's also needed in my opinion.

So, for now, does it just replace the whole selection with a space when we type a space after making a selection?

**Comment by itsaky** [comment]
> If we could indent the selection, it's reasonable that we should also be able to reverse the effect, de-indenting, right?

The closest thing I can think of for this is the ability to add multiple insert handles (cursors) to the editor ([example](https://www.jetbrains.com/help/idea/multicursor.html)). This should be implemented in the editor itself if we ever plan to do so.

> It's also needed in my opinion.

I have never seen this feature in any of the editors I use. It would unnecessarily complicate things if we do implement it.

This won't be worked on.

> So, for now, does it just replace the whole selection with a space when we type a space after making a selection?

Yes.

**Comment by gituser1000000** [comment]
I just remembered the standard shortcuts that I'm used to in some editors that I've used. I thought I might just put a note here.

-User makes a selection.
-User presses 'TAB to move every line of the selection to the right by a 'TAB' (or spaces equivalent to a 'TAB').
-User presses 'SHIFT-TAB' to move every line of the selection to the left by a 'TAB' (or spaces equivalent to 'TAB').
-If there is already at least one line in the selection that reaches the start of the line, 'SHIFT-TAB' should do nothing.

**Comment by itsaky** [comment]
> User presses SHIFT-TAB to move every line of the selection to the left by a TAB (or spaces equivalent to TAB).

This is better, didn't know about this shortcut. I'll work on it in Rosemoe/sora-editor#482.
